modules/sdk/gradle-plugins-workspace/src/test/java/com/liferay/gradle/plugins/workspace/internal/client/extension/ClientExtensionValidatorTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.gradle.plugins.workspace.internal.client.extension;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.nio.file.Files;
+
+import java.util.HashMap;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Anderson Luiz
+ */
+public class ClientExtensionValidatorTest {
+
+	@Test
+	public void testShouldValidateSuccessfullyIfNoFrontendTokenDefinitionProvided() {
+		ClientExtension clientExtension = new ClientExtension();
+
+		clientExtension.type = "themeCSS";
+		clientExtension.typeSettings = new HashMap<String, Object>() {
+		};
+
+		Project project = Mockito.mock(Project.class);
+
+		ClientExtensionValidator clientExtensionValidator =
+			new ClientExtensionValidator();
+
+		try {
+			clientExtensionValidator.validate(clientExtension, project);
+		}
+		catch (Exception exception) {
+			Assert.fail("It should validate successfully");
+		}
+	}
+
+	@Test
+	public void testThrowsUnableToFindFrontendTokenDefinitionFileException() {
+		ClientExtension clientExtension = new ClientExtension();
+
+		clientExtension.type = "themeCSS";
+		clientExtension.typeSettings = new HashMap<String, Object>() {
+			{
+				put("frontendTokenDefinition", "inexistent-file.json");
+			}
+		};
+
+		Project project = Mockito.mock(Project.class);
+
+		Mockito.when(
+			project.file(Mockito.anyString())
+		).thenReturn(
+			new File("")
+		);
+
+		ClientExtensionValidator clientExtensionValidator =
+			new ClientExtensionValidator();
+
+		GradleException gradleException = Assert.assertThrows(
+			GradleException.class,
+			() -> clientExtensionValidator.validate(clientExtension, project));
+
+		String exceptionLocalizedMessage =
+			gradleException.getLocalizedMessage();
+
+		Assert.assertTrue(
+			exceptionLocalizedMessage.contains("Unable to find file"));
+	}
+
+	@Test
+	public void testThrowsUnableToParseFrontendTokenDefinitionFileException()
+		throws IOException {
+
+		ClientExtension clientExtension = new ClientExtension();
+
+		clientExtension.type = "themeCSS";
+		clientExtension.typeSettings = new HashMap<String, Object>() {
+			{
+				put("frontendTokenDefinition", "file.json");
+			}
+		};
+
+		Project project = Mockito.mock(Project.class);
+
+		File file = File.createTempFile("ftd", "json");
+
+		file.deleteOnExit();
+
+		Files.write(file.toPath(), "[][i".getBytes());
+
+		Mockito.when(
+			project.file(Mockito.anyString())
+		).thenReturn(
+			file
+		);
+
+		ClientExtensionValidator clientExtensionValidator =
+			new ClientExtensionValidator();
+
+		GradleException gradleException = Assert.assertThrows(
```

Comment:
We do not use `Assert.assertThrows`, there are no usages in portal. Please do a try-catch.

Reply:
I think we should stick with `Assert.assertThrows`, actually. `try-catch` is a production code, not a test code. Use it in test code was probably a hack to test scenarios with exceptions during earlier versions of JUnit, since there was no other ways to do this. On later versions new ways comes up, like `@Test ( expected = ... )` ( there are many tests with this approach in Liferay) and `@Rule` and finally `assertThrows` , which is simpler and also more legible.

Reply:
You're right, it's simpler and makes more sense. But there is a reason (I don't know which one) we do not use it :/

If the `@Test (expected = ...)` fits your need, go with that. Otherwise we go with the try-catch. Brian will complain about assertThrows :sweat:

The rule of thumb is to check whether there are usages in our code or not.

Running `git grep assertThrows` on `/liferay-portal` returns nothing

Reply:
Yep, this is true. We don't want to add new patterns unless it's absolutely necessary.